optimum/intel/neural_compressor/configuration.py

```diff
@@ -90,3 +92,172 @@ def _create_distillation_config(config: Union[Dict, DistillationConfig]):
+                - "rtn": weight quantization based round to nearest will be applied.
+                - "gptq": A new one-shot weight quantization method based on approximate second-order information,
+                    that is both highly-accurate and highly efficient. The weights of each column are updated based on
+                    the fixed-scale pseudo-quantization error and the inverse of the Hessian matrix calculated from the activations.
+                    The updated columns sharing the same scale may generate a new max/min value,
+                    so the scale needs to be saved for restoration.
+                - "autoround": AutoRound is an advanced weight-only quantization algorithm for low-bits LLM inference.
+                    It's tailored for a wide range of models and consistently delivers noticeable improvements,
+                    often significantly outperforming SignRound with the cost of more tuning time for quantization.
+    """
+
+    def __init__(
+        self,
+        bits: int = 8,
+        sym: bool = False,
+        num_samples: Optional[int] = None,
+        batch_size: Optional[int] = None,
+        seq_len: Optional[int] = None,
+        tokenizer: Optional[str] = None,
+        dataset: Optional[Union[str]] = None,
+        group_size: Optional[int] = None,
+        scale_dtype: Optional[str] = None,
+        compute_dtype: Optional[str] = None,
+        use_layer_wise: Optional[bool] = False,
+        desc_act: Optional[bool] = False,
+        damp_percent: Optional[float] = 0.01,
+        block_size: Optional[int] = 128,
+        static_groups: Optional[bool] = False,
+        true_sequential: Optional[bool] = False,
+        quant_method: Union[str] = "gptq",
+        **kwargs,
+    ):
+        super().__init__(bits=bits, sym=sym, num_samples=num_samples, batch_size=batch_size, seq_len=seq_len)
+        self.tokenizer = tokenizer
+        self.dataset = dataset
+        self.quant_method = quant_method
+        self.group_size = group_size or (-1 if bits == 8 else 128)
+        self.scale_dtype = scale_dtype
+        self.compute_dtype = compute_dtype
+        self.use_layer_wise = use_layer_wise
+
+        self.desc_act = desc_act
+        self.damp_percent = damp_percent
+        self.block_size = block_size
+        self.static_groups = static_groups
+        self.true_sequential = true_sequential
+
+        if self.use_layer_wise:
+            self.model_name_or_path = kwargs.get("model_name_or_path", None)
+            if self.model_name_or_path is None:
+                raise ValueError("model_name_or_path is necessary for layer wise quantization.")
+        self.post_init_gptq()
+
+    def post_init_cpu(self):
+        if self.compute_dtype is not None and self.compute_dtype not in [
+            "fp32",
+            "bf16",
+            "fp16",
+            "int8",
+        ]:
+            raise ValueError("compute_dtype must be 'fp32', 'bf16', 'fp16', 'int8'.")
```

Comment:
would be easier to update / make sure the message error is correct :
```suggestion
        compute_dtype = ["fp32", "bf16", "fp16", "int8"]
        if self.compute_dtype is not None and self.compute_dtype not in ["fp32", "bf16", "fp16", "int8"];
            raise ValueError(f"`compute_dtype` must be in {compute_dtype}")
```

(true for all above)